freqtrade/optimize/backtesting.py

```diff
@@ -73,32 +89,47 @@ def _generate_text_table(self, data: Dict[str, Dict], results: DataFrame) -> str
         headers = ['pair', 'buy count', 'avg profit %',
                    'total profit ' + stake_currency, 'avg duration', 'profit', 'loss']
         for pair in data:
-            result = results[results.currency == pair]
+            result = results[results.pair == pair]
             tabular_data.append([
                 pair,
                 len(result.index),
                 result.profit_percent.mean() * 100.0,
-                result.profit_BTC.sum(),
-                result.duration.mean(),
-                len(result[result.profit_BTC > 0]),
-                len(result[result.profit_BTC < 0])
+                result.profit_abs.sum(),
+                result.trade_duration.mean(),
+                len(result[result.profit_abs > 0]),
+                len(result[result.profit_abs < 0])
             ])
 
         # Append Total
         tabular_data.append([
             'TOTAL',
             len(results.index),
             results.profit_percent.mean() * 100.0,
-            results.profit_BTC.sum(),
-            results.duration.mean(),
-            len(results[results.profit_BTC > 0]),
-            len(results[results.profit_BTC < 0])
+            results.profit_abs.sum(),
+            results.trade_duration.mean(),
+            len(results[results.profit_abs > 0]),
+            len(results[results.profit_abs < 0])
         ])
         return tabulate(tabular_data, headers=headers, floatfmt=floatfmt, tablefmt="pipe")
 
+    def _store_backtest_result(self, recordfilename: Optional[str], results: DataFrame) -> None:
+
+        records = []
+        print(results)
+        for index, trade_entry in results.iterrows():
```

Comment:
Consider using a list comprehension to build `records`. It's usually faster to build lists that way.

Reply:
good point, thanks